app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -244,9 +246,12 @@ class Core(
                 AdsTelemetryMiddleware(adsTelemetry),
                 LastMediaAccessMiddleware(),
                 HistoryMetadataMiddleware(historyMetadataService),
-                SearchTermTabGroupMiddleware(),
                 SessionPrioritizationMiddleware()
```

Comment:
Shouldn't the check be around this Middleware?

Reply:
My bad 😅. I updated the PR. Thanks for catching this.